app/models/manageiq/providers/nuage/network_manager/vsd_client/rest.rb

```diff
@@ -32,46 +32,40 @@ def append_headers(key, value)
     @headers[key] = value
   end
 
+  def reset_headers
+    @headers = {
+      'X-Nuage-Organization' => 'csp',
+      'Content-Type'         => 'application/json; charset=UTF-8'
+    }
+  end
+
   def get(url)
-    if @api_key == ''
-      login
-    end
-    $nuage_log.debug("GET for Nuage VSD url #{url}")
-    RestClient::Request.execute(:method => :get, :url => url, :user => @user, :password => @api_key,
-     :headers => @headers, :verify_ssl => false) do |response|
-      return response
-    end
+    request(url)
   end
 
   def delete(url)
-    if @api_key == ''
-      login
-    end
-    RestClient::Request.execute(:method => :delete, :url => url, :user => @user, :password => @api_key,
-    :headers => @headers, :verify_ssl => false) do |response|
-      return response
-    end
+    request(url, :method => :delete)
   end
 
   def put(url, data)
-    if @api_key == ''
-      login
-    end
-
-    RestClient::Request.execute(:method => :put, :data => data, :url => url, :user => @user, :password => @api_key,
-    :headers => @headers, :verify_ssl => false) do |response|
-      return response
-    end
+    request(url, :method => :put, :data => data)
   end
 
   def post(url, data)
-    if @api_key == ''
-      login
-    end
+    request(url, :method => :post, :data => data)
+  end
 
-    RestClient::Request.execute(:method => :post, :data => data, :url => url, :user => @user, :password => @api_key,
-    :headers => @headers, :verify_ssl => false) do |response|
-      return response
-    end
+  def request(url, method: :get, data: nil, verify: false)
```

Comment:
This is really a nice catch.

Reply:
I suggest you use `verify_ssl` as the name of the argument. Just `verify` is a bit misleading to me.

Reply:
Done